core-war/src/main/webapp/util/javaScript/vuejs/components/admin/silverpeas-space-and-component-browser.js

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright (C) 2000 - 2024 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+//# sourceURL=/silverpeas/util/javaScript/vuejs/components/admin/silverpeas-space-and-component-browser.js
+
+(function() {
+
+  const templateRepository = new VueJsAsyncComponentTemplateRepository(webContext +
+      '/util/javaScript/vuejs/components/admin/silverpeas-space-and-component-browser-templates.jsp');
+
+  const SpaceAndComponentManager = function(service) {
+    let roots;
+    this.getRoots = function() {
+      if (!roots) {
+        return service.listChildrenByIdOrUri('').then(function(spaces) {
+          roots = spaces;
+          return spaces;
+        });
+      }
+      return sp.promise.resolveDirectlyWith(roots);
+    }
+  }
+
+  window.SCB_MAIN_MIXIN = {
```

Comment:
There is two new VueJS components: `silverpeas-space-and-component-browser.js` and `silverpeas-space-and-component-selector.js`.
In some context, publication paths management in kmelia for example, just the browser is needed. In other contexts, both are needed. In order to avoid to fetch unecessary assets when just the browser is needed, the components have been implemented in different sources.
As `silverpeas-space-and-component-browser.js` exposes at least the same props than silverpeas-space-and-component-browser.js` I have exposed the MIXIN.
MIXIN in VueJS are used to centralize common declarations and behaviors and manager component behavior ovveriding.

Reply:
Ok but this mixin is also redefined in other js and in their own context. Is it normal?